spec/spec_helper.rb

```diff
@@ -75,3 +81,26 @@ def in_tmp_dir
     end
   end
 end
+
+# Executes the given block with a temporary file with the given `file_name`
+def with_tmp_file_path_for_file_named(file_name)
+  in_tmp_dir do |tmp_dir|
+    file_path = File.join(tmp_dir, file_name)
+
+    File.write(file_path, '')
+    yield file_path
+    File.delete(file_path)
```

Comment:
You want to make sure the intermediate file is deleted _even if_ the code crashes in the middle of the execution:

```suggestion
  ensure
    File.delete(file_path)
```

PS: I'm not _fully sure_ that we can use `ensure` if we don't even have a `rescue` in-between… but I think that's valid?

PPS: I actually thought about writing such a helper at first, but wasn't sure it was worth it 🤷 . I also thought about using `Tempfile.create` which is dedicated to a very similar purpose… except that `Tempfile` does not let you enforce a specific filename (and instead create the temp file in `/tmp` and find a unique basename, instead of making the uniqueness by using an intermediate temp dir), which wouldn't have worked for our use case given the SHA1 is dependent on the basename used in the unit test…